src/Comparator/BackwardsCompatibility/PropertyBased/PropertyVisibilityReduced.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\ApiCompare\Comparator\BackwardsCompatibility\PropertyBased;
+
+use Roave\ApiCompare\Change;
+use Roave\ApiCompare\Changes;
+use Roave\BetterReflection\Reflection\ReflectionProperty;
+use function sprintf;
+
+final class PropertyVisibilityReduced implements PropertyBased
+{
+    private const VISIBILITY_PRIVATE = 'private';
+
+    private const VISIBILITY_PROTECTED = 'protected';
+
+    private const VISIBILITY_PUBLIC = 'public';
+
+    public function compare(ReflectionProperty $fromProperty, ReflectionProperty $toProperty) : Changes
+    {
+        $visibilityFrom = $this->propertyVisibility($fromProperty);
+        $visibilityTo   = $this->propertyVisibility($toProperty);
+
+        if ($visibilityFrom <= $visibilityTo) {
+            return Changes::new();
+        }
+
+        return Changes::fromArray([Change::changed(
+            sprintf(
+                'Property %s#$%s visibility reduced from %s to %s',
```

Comment:
Just a minor thing, but do we have consistency with using `#` vs `::`? What are the rules here for formatting?

Reply:
I'm trying to use `#` for instance, `::` for statics. I'll extract this to a helper, much like for the function naming.